sdk/storage/storage-file/README.md

```diff
@@ -234,25 +249,78 @@ async function streamToString(readableStream) {
     readableStream.on("error", reject);
   });
 }
+```
+
+### Download a file and convert it to a string (Browsers)
+
+```javascript
+  // Get file content from position 0 to the end
+  // In browsers, get downloaded data by accessing downloadFileResponse.blobBody
+  const downloadFileResponse = await fileClient.download(0);
+  console.log(
+    `Downloaded file content${await streamToString(
+      downloadFileResponse.blobBody
+    )}`
+  );
 
-// An async method returns a Promise object, which is compatible with then().catch() coding style.
-main()
-  .then(() => {
-    console.log("Successfully executed sample.");
-  })
-  .catch(err => {
-    console.log(err.message);
+// [Browser only] A helper method used to convert a browser Blob into string.
+export async function blobToString(blob: Blob): Promise<string> {
+  const fileReader = new FileReader();
+  return new Promise<string>((resolve, reject) => {
+    fileReader.onloadend = (ev: any) => {
+      resolve(ev.target!.result);
+    };
+    fileReader.onerror = reject;
+    fileReader.readAsText(blob);
   });
+}
 ```
 
-## More Code Samples
+A complete example of basic scenarios is at [samples/basic.js](https://github.com/azure/azure-sdk-for-js/tree/master/sdk/storage/storage-file/samples/basic.js).
 
-- [File Storage Examples](https://github.com/azure/azure-storage-js/tree/master/file/samples)
-- [File Storage Examples - Test Cases](https://github.com/azure/azure-storage-js/tree/master/file/test/)
+## Troubleshooting
 
-## License
+It could help diagnozing issues by turning on the console logging. Here's an example logger implementation. First add a custom logger:
 
-This project is licensed under MIT.
+```javascript
+class ConsoleHttpPipelineLogger {
+  constructor(minimumLogLevel) {
+    this.minimumLogLevel = minimumLogLevel;
+  }
+  log(logLevel, message) {
+    const logMessage = `${new Date().toISOString()} ${HttpPipelineLogLevel[logLevel]}: ${message}`;
+    switch (logLevel) {
+      case HttpPipelineLogLevel.ERROR:
+        console.error(logMessage);
+        break;
+      case HttpPipelineLogLevel.WARNING:
+        console.warn(logMessage);
+        break;
+      case HttpPipelineLogLevel.INFO:
+        console.log(logMessage);
+        break;
+    }
+  }
+}
+```
+
+When creating the `FileServiceClient` instance, pass the logger in the options
+
+```javascript
+  const fileServiceClient = new FileServiceClient(
+    `https://${account}.file.core.windows.net`,
+    sharedKeyCredential, {
+      logger: new ConsoleHttpPipelineLogger(HttpPipelineLogLevel.INFO),
+    }
+  );
+```
+
+## Next steps
+
+More code samples
+
+- [File Storage Examples](https://github.com/azure/azure-storage-js/tree/master/file/samples)
+- [File Storage Examples - Test Cases](https://github.com/azure/azure-storage-js/tree/master/file/test/)
```

Comment:
Update links

Reply:
wait you didn't do it!?

---
In reply to: [295985777](https://github.com/Azure/azure-sdk-for-js/pull/3744#discussion_r295985777) [](ancestors = 295985777)

Reply:
Might have missed! 😃